chain/hyperliquid/client/client.go

```diff
@@ -0,0 +1,615 @@
+	txInput.HyperliquidChain = client.HyperliquidChain
+
+	return txInput, nil
+}
+
+// Deprecated method - use FetchTransferInput
+func (client *Client) FetchLegacyTxInput(ctx context.Context, from xc.Address, to xc.Address) (xc.TxInput, error) {
+	// No way to pass the amount in the input using legacy interface, so we estimate using min amount.
+	chainCfg := client.Asset.GetChain().Base()
+	args, _ := xcbuilder.NewTransferArgs(chainCfg, from, to, xc.NewAmountBlockchainFromUint64(1))
+	return client.FetchTransferInput(ctx, args)
+}
+
+// SubmitTx submits a hyperliquid tx
+func (client *Client) SubmitTx(ctx context.Context, tx xc.Tx) error {
+	payload, err := tx.Serialize()
+	if err != nil {
+		return fmt.Errorf("failed to serialize transaction: %w", err)
+	}
+
+	response, err := client.CallExchange(ctx, payload)
+	if err != nil {
+		return fmt.Errorf("failed to post transaction: %w", err)
+	}
+
+	if !response.IsOk() {
+		return fmt.Errorf("failed to submit tx: %s", response.Response)
+	}
+
+	return nil
+}
+
+// Returns transaction info - legacy/old endpoint
+func (client *Client) FetchLegacyTxInfo(ctx context.Context, txHash xc.TxHash) (xclient.LegacyTxInfo, error) {
+	return xclient.LegacyTxInfo{}, errors.New("not implemented")
+}
+
+func (client *Client) fetchTxDetails(ctx context.Context, hash string) (types.Transaction, error) {
+	type response struct {
+		Type        string            `json:"type"`
+		Transaction types.Transaction `json:"tx"`
+	}
+
+	var txDetails response
+	err := client.CallExplorer(ctx, MethodTxDetails, map[string]any{
+		"hash": hash,
+	}, &txDetails)
+	return txDetails.Transaction, err
+
+}
+
+func (client *Client) fetchTransactionFee(ctx context.Context, address xc.Address, hash xc.TxHash) (xc.AmountHumanReadable, string, error) {
+	var response []types.UserNonFundingLedgerUpdate
+	err := client.CallInfo(ctx, MethodUserNonFundingLedgerUpdates, map[string]any{
+		"user": address,
+	}, &response)
+	if err != nil {
+		return xc.AmountHumanReadable{}, "", fmt.Errorf("failed to fetch user fees: %w", err)
+	}
+
+	for _, update := range response {
+		if update.Hash == string(hash) {
+			fee := update.GetFee()
+			feeHr, err := xc.NewAmountHumanReadableFromStr(fee)
+			if err != nil {
+				return xc.AmountHumanReadable{}, "", fmt.Errorf("failed to parse amount human readable: %w", err)
+			}
+
+			token := update.GetFeeToken()
+
+			return feeHr, token, nil
+		}
+	}
+
+	return xc.AmountHumanReadable{}, "", fmt.Errorf("coudln't find tx %s in user ledger updates", hash)
+}
+
+// Traditional "FetchTxInfo" implementation - use a native Hyperliquid transaction hash for transaction
+// info lookup
+func (client *Client) fetchTxInfoByHash(ctx context.Context, args *txinfo.Args) (xclient.TxInfo, error) {
+	contract, _ := args.Contract()
+	txDetails, err := client.fetchTxDetails(ctx, string(args.TxHash()))
+	if err != nil {
+		return xclient.TxInfo{}, fmt.Errorf("failed to fetch tx details: %w", err)
+	}
```

Comment:
should have a case for when transaction is not found, e.g. see on other chains:
https://github.com/cordialsys/crosschain/blob/main/chain/solana/client/client.go#L300-L306